tests/src/python/providertestbase.py

```diff
@@ -1206,3 +1208,19 @@ def testGeneratedColumns(self):
         feature = vl.getFeature(10)
         self.assertTrue(feature.isValid())
         self.assertEqual(feature.attribute(1), "test:10")
+
+        # Test insertion with default value evaluation on provider side to be sure
+        # it doesn't fail generated columns
+        vl.dataProvider().setProviderProperty(QgsDataProvider.EvaluateDefaultValues, True)
+
+        vl.startEditing()
+        feature = QgsVectorLayerUtils.createFeature(vl, QgsGeometry(), {0: 8})
+        vl.addFeature(feature)
+        self.assertTrue(feature.id() < 0)
+        # to be fixed
```

Comment:
?

Reply:
It's related to #38032 . Management of read-only fields has to be improved to uncomment this line. I propose to wait for  #38032 to be merged and I'll rebase/uncomment this line.

And I'll backport manually without it in 3.16